src/Infrastructure/Services/Development/NullEmailService.cs

```diff
@@ -1,16 +1,36 @@
-﻿using System;
-using System.Threading.Tasks;
+﻿using System.Threading.Tasks;
 using Application.Contracts;
+using Application.Settings;
+using Microsoft.Extensions.Logging;
 
-namespace Infrastructure.Services.Development
+namespace Infrastructure.Services.Development;
+
+public class NullEmailService : IEmailService
 {
-    public class NullEmailService : IEmailService
+    private readonly ILogger<NullEmailService> _logger;
+    private readonly IEmailService _emailService;
+    private readonly DevSettings _devSettings;
+
+    public NullEmailService(ILogger<NullEmailService> logger, IEmailService emailService, DevSettings devSettings)
+    {
+        _logger = logger;
+        _emailService = emailService;
+        _devSettings = devSettings;
+    }
+
+    public Task SendEmailAsync(string to, string subject, string body)
     {
-        public Task SendEmailAsync(string to, string subject, string body)
+        if (_devSettings.ShouldBypassEmail)
```

Comment:
The default should be off in non-production and you have to turn it on, rather than the other way around.